webrtc/videoroom/assets/src/membraneWebRTC.ts

```diff
@@ -65,10 +65,10 @@ export class MembraneWebRTC {
   }
 
   public addTrack = (track: MediaStreamTrack, stream: MediaStream) => {
-    if(this.connection) {
+    if (this.connection) {
       throw new Error("Adding tracks when connection is established is not yet supported");
     }
-    this.localTracks.add(track);
+    this.localTracks.add([track, stream]);
```

Comment:
Are you sure that it works? I believe that javascript compares that elements by reference so you will never end up with unique values. 

```js
const s = new Set();
s.add([1,2]);
s.add([1,2]);
```
This set will have 2 entries instead of 1 because array references are compared.